assembl/static2/js/app/components/administration/discussion/languageAdmin.jsx

```diff
@@ -32,6 +32,11 @@ class LanguageSection extends React.Component {
     };
   }
 
+  componentWillMount() {
+    //Hide the admin language menu
+    this.props.toggleLanguageMenu(false);
```

Comment:
I don't think that's the best way of doing it, but I don't have enough context by just reading the code about expected behaviour and possible other solutions

Reply:
also this commit does too many different things, please try to separate more

Reply:
We don't know in advance which administration sections will show the language menu. So, I let lifted the state of show/hide to the parent-most component, and as a prop, I passed a method that allows each child administration component to determine if they desire the language menu or not. I tried to follow the best recommendation here https://facebook.github.io/react/docs/lifting-state-up.html.